app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -21,11 +23,39 @@
     private DeviceUtils() {
     }
 
+    /**
+     * Checks if the screen layout was forced to the given layout.
+     * @param context Application context
+     * @param resId The resource ID of the wanted screen layout
+     * @return true if the layout was forced, false if the layout was forced to something else,
+     * and null if the screen layout is auto
+     */
+    private static Boolean layoutForcedTo(final Context context, final int resId) {
```

Comment:
`boolean` and `@StringRes`
```suggestion
    private static boolean layoutForcedTo(final Context context, @StringRes final int resId) {
```
I don't like the fact that this function could return `null`. It should return `true` if the layout is forced to the requested value and `false` othewise.

Reply:
but that is required below to distinguish correctly.

I suggest to annotate the method with `@Nullable`

Reply:
Mmmh ok, but I really don't like giving a contextual meaning to `null`. A better thing would be to implement a `getScreenLayoutMode()` function which returns an enum value which can be `AUTO`, `FORCE_TV`, `FORCE_TABLET`, `FORCE_PHONE`, and then let the caller decide what to do with them. What do you think?

Reply:
Yes, that sounds like a good solution